src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,62 @@
+.calendar {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+
+  &__month {
+    width: $calendar-width;
+    display: flex;
+    flex-wrap: wrap;
+    justify-content: left;
+    gap: $gap-size;
+  }
+
+  &__day {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    background-color: #eee;
+    border: 1px solid #000;
+    @include square($box-size);
+    transition: transform 0.5s, background-color 0.5s;
+  }
+
+  &__day:hover {
+    cursor: pointer;
+    background-color: pink;
+    transform: translateY(-20px);
+  }
+}
+
+@for $data-number from 1 through 31 {
+  .calendar__day--#{$data-number}::before {
+    content: "#{$data-number}";
+    font-size: 30px;
+  }
+}
+
+$start-day: (
+  "mon": 0 * $first-day-margin,
+  "tue": 1 * $first-day-margin,
+  "wed": 2 * $first-day-margin,
+  "thu": 3 * $first-day-margin,
+  "fri": 4 * $first-day-margin,
+  "sat": 5 * $first-day-margin,
+  "sun": 6 * $first-day-margin
```

Comment:
store only day name and number here, you shouldn't repeat some formula many times
go through this variable in cycle and set correct margins (using formula) inside the cycle one time